shell/browser/mac/in_app_purchase_observer.mm

```diff
@@ -94,6 +94,37 @@ - (NSString*)dateToISOString:(NSDate*)date {
   return [dateFormatter stringFromDate:date];
 }
 
+/**
+ * Convert a SKPaymentDiscount object to a PaymentDiscount structure.
+ *
+ * @param paymentDiscount - The SKPaymentDiscount object to convert.
+ */
+- (in_app_purchase::PaymentDiscount)skPaymentDiscountToStruct:
+    (SKPaymentDiscount*)paymentDiscount API_AVAILABLE(macosx(10.14.4)) {
+  in_app_purchase::PaymentDiscount paymentDiscountStruct;
+
+  if (paymentDiscount.identifier != nil) {
+    paymentDiscountStruct.identifier = [paymentDiscount.identifier UTF8String];
+  }
```

Comment:
I am getting an error after removing the `nil` check originating from `skPaymentToStruct` which I believe was thrown because I removed the `nil` check for `paymentStruct.applicationUsername = [payment.applicationUsername UTF8String];` assignment 🤔 

I've re-added the `nil` check back to properties that are optional based on apple's doc.

```
../../buildtools/third_party/libc++/trunk/include/string:2551: _LIBCPP_ASSERT '__s != nullptr' failed. string::assign received nullptr
Received signal 6
0   Electron Framework                  0x0000000111ad6529 base::debug::CollectStackTrace(void**, unsigned long) + 9
1   Electron Framework                  0x00000001119fc243 base::debug::StackTrace::StackTrace() + 19
2   Electron Framework                  0x0000000111ad6481 base::debug::(anonymous namespace)::StackDumpSignalHandler(int, __siginfo*, void*) + 1409
3   libsystem_platform.dylib            0x00007fff204d7d7d _sigtramp + 29
4   libdyld.dylib                       0x00007fff204acce8 _dyld_fast_stub_entry(void*, long) + 65
5   libsystem_c.dylib                   0x00007fff203e7406 abort + 125
6   Electron Framework                  0x0000000116e18132 std::__1::__libcpp_abort_debug_function(std::__1::__libcpp_debug_info const&) + 82
7   Electron Framework                  0x000000010d6764cf std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::operator=(char const*) + 79
8   Electron Framework                  0x000000010d6766b0 -[InAppTransactionObserver skPaymentToStruct:] + 240
9   Electron Framework                  0x000000010d676ab0 -[InAppTransactionObserver skPaymentTransactionToStruct:] + 736
10  Electron Framework                  0x000000010d675fe5 -[InAppTransactionObserver runCallback:] + 341
11  StoreKit                            0x00007fff5403e617 __48-[SKPaymentQueue _updatedTransactions:restored:]_block_invoke_2 + 62
12  libdispatch.dylib                   0x00007fff202e7623 _dispatch_call_block_and_release + 12
13  libdispatch.dylib                   0x00007fff202e8806 _dispatch_client_callout + 8
14  libdispatch.dylib                   0x00007fff202eae37 _dispatch_queue_override_invoke + 775
15  libdispatch.dylib                   0x00007fff202f7818 _dispatch_root_queue_drain + 326
16  libdispatch.dylib                   0x00007fff202f7f70 _dispatch_worker_thread2 + 92
17  libsystem_pthread.dylib             0x00007fff2048f417 _pthread_wqthread + 244
18  libsystem_pthread.dylib             0x00007fff2048e42f start_wqthread + 15
[end of stack trace]
```